src/common/components/repositories-list/index.js

```diff
@@ -69,26 +70,28 @@ class RepositoriesList extends Component {
     }
   }
 
-  renderRow(snap) {
+  renderRow(snap, index) {
+    const { hasNoRegisteredNames, registerName, snapBuilds, authStore } = this.props;
     const { fullName } = parseGitHubRepoUrl(snap.git_repository_url);
 
     let latestBuild = null;
-    const snapBuilds = this.props.snapBuilds[fullName];
+    const currentSnapBuilds = snapBuilds[fullName];
 
-    if (snapBuilds && snapBuilds.success && snapBuilds.builds.length) {
-      latestBuild = snapBuilds.builds[0];
+    if (currentSnapBuilds && currentSnapBuilds.success && currentSnapBuilds.builds.length) {
+      latestBuild = currentSnapBuilds.builds[0];
     }
 
-    const registerNameStatus = this.props.registerName[fullName] || {};
+    const registerNameStatus = registerName[fullName] || {};
 
     return (
       <RepositoryRow
         key={ `repo_${fullName}` }
         snap={ snap }
         latestBuild={ latestBuild }
         fullName={ fullName }
-        authStore={ this.props.authStore }
+        authStore={ authStore }
         registerNameStatus={ registerNameStatus }
+        registerNameIsOpen={ !index && hasNoRegisteredNames }
       />
     );
   }
```

Comment:
I know it's not part of this PR, but the `onButtonClick` method just below that doesn't seem to be used anywhere and looks like it was just copied from somewhere else.
Also `createSnap` action dispatched from it shouldn't be needed at all in this component.